fofix/core/GameEngine.py

```diff
@@ -292,29 +289,94 @@ def __init__(self, config = None):
 
         self.setlistMsg = None
 
+        # Init theme system
+        themename = self.config.get("coffee", "themename")
+        themepath  = os.path.join(Version.dataPath(), "themes", themename)
+        self._initTheme(themename, themepath)
 
         # Load game modifications
         Mod.init(self)
-        self.task.addTask(self.input, synced = False)
 
+        self.task.addTask(self.input, synced = False)
         self.task.addTask(self.view, synced = False)
-
         self.task.addTask(self.resource, synced = False)
 
         self.data = Data(self.resource, self.svg)
 
+        self.input.addKeyListener(FullScreenSwitcher(self), priority = True)
+        self.input.addSystemEventListener(SystemEventHandler(self))
+
+        self.debugLayer         = None
+        self.startupLayer       = None
+        self.loadingScreenShown = False
+        self.graphicMenuShown   = False
+
+        log.debug("Ready.")
+
+
+    # evilynux - This stops the crowd cheers if they're still playing (issue 317).
+    def quit(self):
+        # evilynux - self.audio.close() crashes when we attempt to restart
+        if not self.restartRequested:
+            self.audio.close()
+        Player.savePlayers()
+        for taskData in list(self.task.tasks):
+            self.task.removeTask(taskData['task'])
+        self.running = False
+
+
+    def _initTheme(self, themename, themepath):
```

Comment:
Something else to mention here. I'm not a fan of _ prefix's on methods. I understand this is something that will be used at init time hence the function name no need to underscore it.